R/impact_central.R

```diff
@@ -24,27 +24,32 @@ get_raw_impact_details <- function(con, meta1, burden_outcome, is_under5 = FALSE
   v <- determine_vaccine_delivery(meta1)
   i <- unique(grepl("routine", v))
   j <- unique(grepl("campaign", v))
-  if(i & j & meta1$method[1] == "method2a"){
+  if(any(i) & any(j) & meta1$method[1] == "method2a"){
     stop("method2a is vaccine delivery specific, cannot include both routine and campaign impact at the same time.")
   }
 
   # constain db extraction by country and age
-  countries_to_extract <- DBI::dbGetQuery(con, sprintf("SELECT nid FROM country WHERE id IN %s",
-                                                       sql_in(countries_to_extract)))
+
   age_constrain <- ifelse(is_under5, "AND age < 5", "") #applies to all methods
-  country_constrain <- ifelse(is.null(countries_to_extract), "",
-                              sprintf("AND country IN %s", sql_in(countries_to_extract, text_item = FALSE)))
+
+  if (is.null(countries_to_extract)){
+    country_constrain <- ""
+  } else{
+    countries_to_extract <- DBI::dbGetQuery(con, sprintf("SELECT nid FROM country WHERE id IN %s",
+                                                         sql_in(countries_to_extract)))
+    country_constrain <- sprintf("AND country IN %s", sql_in(countries_to_extract, text_item = FALSE))
+  }
 
   # set up db extraction sql queries
-  if((meta1$method[1] == "method0") | (meta1$method[1] == "method2a" & j)){
+  if((meta1$method[1] == "method0") | (meta1$method[1] == "method2a" & any(j))){
```

Comment:
Same with this one here - I think should use `||` and `&&`

Reply:
This is done.